internal/memdb/memdb.go

```diff
@@ -165,12 +181,12 @@ func (c *client) Sync(_ context.Context, options plugin.SyncOptions, res chan<-
 	return nil
 }
 
-func (c *client) Tables(context.Context, plugin.TableOptions) (schema.Tables, error) {
+func (c *client) Tables(ctx context.Context, opts plugin.TableOptions) (schema.Tables, error) {
 	tables := make(schema.Tables, 0, len(c.tables))
 	for _, table := range c.tables {
 		tables = append(tables, table)
 	}
-	return tables, nil
+	return tables.FilterDfs(opts.Tables, opts.SkipTables, opts.SkipDependentTables)
```

Comment:
This is needed so we set parents implicitly via `Copy` (see also https://github.com/cloudquery/cloudquery/issues/14637)